### PR TITLE
feat: introduce style name API

### LINKED ISSUE #37: Add API stylenames AbstractComponent
<html>
<body>
<google-sheets-html-origin>

return| method 
-- | --
void | addStyleName(String style)Adds one or more style names to this component.
default void | addStyleNames(String... styles)Adds one or more style names to this component by using one or multiple parameters.
String | getPrimaryStyleName()Gets the primary style name of the component.
String | getStyleName()Gets all user-defined CSS style names of a component.
void | removeStyleName(String style)Removes one or more style names from component.
default void | removeStyleNames(String... styles)Removes one or more style names from component.
void | setPrimaryStyleName(String style)Changes the primary style name of the component.
void | setStyleName(String style)Sets one or more user-defined style names of the component, replacing any previous user-defined styles.
default void | setStyleName(String style, boolean add)Adds or removes a style name.

</body>
</html>

Update the HL+VL to set their primary style names as `v-horizontal-layout` + `v-vertical-layout`.

### PR DESCRIPTION
**Pull request by pleku**
Creates all the styles on server side and writes them to the element on
beforeClientResponse-phase. As in addition to the primary style name,
either the internal or user defined styles need to be stored in the
component, so for simplicity, everything is stored. 

Fixes #37

### CODE REVIEW

File: vaadin-legacy-components-flow/src/test/java/com/vaadin/legacy/ui/AbstractOrderedLayoutTest.java [hunk 1]
```diff
@@ -282,8 +282,14 @@ public void shouldMoveComponentsWithExpandWrapper() {
 
     @Test
     public void shouldHaveCorrectClasses() {
-        Assert.assertTrue(layout.getClassNames().contains("v-layout"));
-        Assert.assertTrue(layout.getClassNames().contains("v-widget"));
+        layout.beforeClientResponse(false);
+        Assert.assertEquals("v-widget v-layout v-horizontal",
+                layout.getElement().getAttribute("class"));
+
+        layout = new AbstractOrderedLayoutImpl(true);
```
Comment (fhdhsni) [inline vaadin-legacy-components-flow/src/test/java/com/vaadin/legacy/ui/AbstractOrderedLayoutTest.java:289]: nit: use `verticalLayout` that already exist in the scope

**Comment by fhdhsni** [comment]
LGTM

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java [hunk 2]
```diff
@@ -261,4 +279,323 @@ protected void doSetHeight(SizeWithUnit height) {
...
+     * The primary style name identifies the component when applying the CSS
+     * theme to the Component. By changing the style name all CSS rules targeted
+     * for that style name will no longer apply, and might result in the
+     * component not working as intended.
+     * </p>
+     *
+     * <p>
+     * To preserve the original style of the component when changing to a new
+     * primary style you should make your new primary style inherit the old
+     * primary style using the SASS @include directive. See more in the SASS
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java]: Definitely, lets bring back the SASS compiler

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java [hunk 3]
```diff
@@ -261,4 +279,323 @@ protected void doSetHeight(SizeWithUnit height) {
...
+        }
+
+        if (style.contains(" ")) {
+            // Split space separated style names and add them one by one.
+            StringTokenizer tokenizer = new StringTokenizer(style, " ");
+            while (tokenizer.hasMoreTokens()) {
+                addStyleName(tokenizer.nextToken());
+            }
+            return;
+        } else {
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java]: Done

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java [hunk 4]
```diff
@@ -261,4 +279,323 @@ protected void doSetHeight(SizeWithUnit height) {
...
+     *
+     * <pre>
+     * Label label = new Label(&quot;This text has style&quot;);
+     * label.addStyleName(&quot;mystyle&quot;);
+     * </pre>
+     *
+     * <p>
+     * Each style name will occur in two versions: one as specified and one that
+     * is prefixed with the style name of the component. For example, if you
+     * have a {@code Button} component and give it "{@code mystyle}" style, the
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java]: Done

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java [hunk 5]
```diff
@@ -261,4 +279,323 @@ protected void doSetHeight(SizeWithUnit height) {
...
+     *
+     * <pre>
+     * Label label = new Label(&quot;This text has a lot of style&quot;);
+     * label.setStyleName(&quot;myonestyle myotherstyle&quot;);
+     * </pre>
+     *
+     * <p>
+     * Each style name will occur in two versions: one as specified and one that
+     * is prefixed with the style name of the component. For example, if you
+     * have a {@code Button} component and give it "{@code mystyle}" style, the
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponent.java]: Done

File: vaadin-legacy-components-flow/src/test/java/com/vaadin/legacy/ui/AbstractOrderedLayoutTest.java (hunk 1 above)
Comment (pleku) [reply vaadin-legacy-components-flow/src/test/java/com/vaadin/legacy/ui/AbstractOrderedLayoutTest.java]: Done

**Review by DiegoCardoso — APPROVED** [review]
🎉

**Review by fhdhsni — APPROVED** [review]

**Pull request merged**